API Sample Refactor (#1215)
spectrocloud/librarium: content/api/2-samples.mdx

* The respective language runtime and package manager installed on your machine. 

* [jq](https://stedolan.github.io/jq/download/) - A command-line JSON processor.

## Configure Scope and Authentication

Variables shown in the following examples are used to interact with the Palette API. The values for these variables are specific to your environment.
The project ID is the Unique Identifier (UID) of the project in which you want to create the cluster. You can find the project ID at top right in the Palette dashboard.

<br />

```shell
export API_KEY="Your API Key"
```
<br />

```shell
export PROJECT_ID="Your Project UID"
```

Some of the endpoints require a cluster ID.

<InfoBox>

If you do not provide the ProjectUid header, then the assumed scope is of the tenant.

</InfoBox>

# Deploy a Cluster

You can use the following endpoint to deploy a cluster. The provider value represents the cloud provider on which you want to deploy the cluster, such as public cloud or on-prem.

**Endpoint**: `https://api.spectrocloud.com/v1/spectroclusters/{provider}`

Set the provider as an environment variable. 

```shell
export PROVIDER="Your Provider"
```

> The payload for creating a host cluster can be lengthy due to the various configurations available for a host cluster. To improve readbility, the payload is condensed in the following examples and lack required values. 

Author: ```suggestion
The payload for creating a host cluster can be lengthy due to the various configurations available for a host cluster. The payload is condensed in the following examples to improve readability and lack all the required values. 
```

icon: ""
hideToC: true
fullWidth: false
hiddenFromNav: false
hideToCSidebar: true
---
import {Intro, IntroButtons} from "shared/components"
import InfoBox from "shared/components/InfoBox"
import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';

# Overview

This workflow demonstrates how to use Spectro Cloud API. You can use the API to automate the provisioning of Kubernetes clusters and applications on Spectro Cloud. 

The API is a RESTful API that uses JSON for serialization and supports HTTP Basic Authentication and token authentication. The API is available at `https://api.spectrocloud.com`.

Use the following examples to familiarize yourself with the API.

<br />

## Prerequisites

* You must have a Spectro Cloud account. If you do not have an account, you can create one at [https://console.spectrocloud.com](https://console.spectrocloud.com).

* An Authentication header with a token value or an API Key with an ApiKey value. 
Learn more about authentication credentials by reviewing the [authentication methods](https://docs.spectrocloud.com/user-management/user-authentication) resource.

> * You must have a Spectro Cloud cluster. You can create a cluster from the Spectro Cloud console.

Reviewer: ```suggestion
* You must have a Palette cluster. You can create a cluster in Palette.
```

* The respective language runtime and package manager installed on your machine. 

* [jq](https://stedolan.github.io/jq/download/) - A command-line JSON processor.

## Configure Scope and Authentication

> In the following examples, the following variables are used to interact with the Palette API. The values for these variables are specific to your environment.

Reviewer: ```suggestion
Variables shown in the following examples are used to interact with the Palette API. The values for these variables are specific to your environment.
```

	}
	defer resp.Body.Close()

	// Read the response body
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Error reading HTTP response:", err)
		return
	}

	// Print the response body
	fmt.Println(string(body))
}

```

</Tabs.TabPane>

<Tabs.TabPane tab="Python" key="node-status-python">

```python
import os
import requests

# Get API key, project ID, and cluster ID from environment variables
api_key = os.environ.get("API_KEY")
project_id = os.environ.get("PROJECT_ID")
cluster_id = os.environ.get("CLUSTER_ID")

# Build the request URL
url = f"https://api.spectrocloud.com/v1/spectroclusters/{cluster_id}?ProjectUid={project_id}"

# Define headers for the request
headers = {
  "Accept": "application/json",
  "ApiKey": api_key
}

# Send the HTTP request using the requests library
response = requests.get(url, headers=headers)

# Check the response status code
if response.status_code == 200:
  # Print the response data (JSON format)
  print("Response data:", response.json())
else:
  # Print the error message if the request failed
  print("Request failed with status code:", response.status_code)
```

</Tabs.TabPane>

<Tabs.TabPane tab="Rust" key="node-status-rust">

> ```

Reviewer: Is rust needed here?
```rust

    let cluster_id = env::var("CLUSTER_ID").unwrap();

    // Build the request URL
    let url = format!("https://api.spectrocloud.com/v1/spectroclusters/{}?ProjectUid={}", cluster_id, project_id);

    // Define headers for the request
    let mut headers = HeaderMap::new();
    headers.insert(ACCEPT, HeaderValue::from_static("application/json"));
    headers.insert(APIKEY, HeaderValue::from_str(&api_key).unwrap());

    // Send the HTTP request using the reqwest crate
    let response = reqwest::Client::new()
        .get(&url)
        .headers(headers)
        .send()
        .await?;

    // Check the response status code
    if response.status().is_success() {
        // Print the response data (JSON format)
        let body = response.text().await?;
        println!("Response data: {}", body);
    } else {
        // Print the error message if the request failed
        println!("Request failed with status code: {}", response.status());
    }

    Ok(())
}
```

</Tabs.TabPane>

</Tabs>

##  Retrieve the Cluster Cloud Config Identifier

The cloud config identifier is a unique identifier for the cloud config that is used to provision the cluster. 

> The values are found in the `spec.cloudConfigRef` field of the cluster object. You can use the following code snippet to retrieve the cluster cloud config identifier `uid` and `kind`. 

Reviewer: ```suggestion
The values are found in the `spec.cloudConfigRef` field of the cluster object. You can use the following code snippet to retrieve the `uid` and `kind` for the cluster cloud config identifier. 
```

}
```

</Tabs.TabPane>

<Tabs.TabPane tab="Python" key="config-identifier-python">

```python
import os
import requests

# Get API key, project ID, and cluster ID from environment variables
api_key = os.environ.get("API_KEY")
project_id = os.environ.get("PROJECT_ID")
cluster_id = os.environ.get("CLUSTER_ID")

# Build the request URL
url = f"https://api.spectrocloud.com/v1/spectroclusters/{cluster_id}?ProjectUid={project_id}"

# Define headers for the request
headers = {
  "Accept": "application/json",
  "ApiKey": api_key
}

# Send the HTTP request using the requests library
response = requests.get(url, headers=headers)

# Check the response status code
if response.status_code == 200:
  # Parse the response JSON and extract the values
  response_json = response.json()
  kind = response_json['spec']['cloudConfigRef']['kind']
  uid = response_json['spec']['cloudConfigRef']['uid']

  # Print the extracted values
  print(f"Kind: {kind}")
  print(f"UID: {uid}")
else:
  # Print the error message if the request failed
  print("Request failed with status code:", response.status_code)
```
</Tabs.TabPane>

<Tabs.TabPane tab="Rust" key="config-identifier-rust">

> ```

Reviewer: Is ``rust`` after the three tick marks?
